packages/runtime/container-runtime/src/dataStores.ts

```diff
@@ -573,11 +574,15 @@ export class DataStores implements IDisposable {
 			Array.from(this.contexts)
 				.filter(([_, context]) => {
 					// Summarizer works only with clients with no local changes. A data store in attaching
-					// state indicates an op was sent to attach a local data store.
-					assert(
-						context.attachState !== AttachState.Attaching,
-						0x588 /* Local data store detected in attaching state during summarize */,
-					);
+					// state indicates an op was sent to attach a local data store, and the the attach op
+					// had not yet round tripped back to the client.
+					if (context.attachState === AttachState.Attaching) {
+						const error = DataProcessingError.create(
```

Comment:
Let's put the assert code in a comment, something like this, for easier searching (and below)

```suggestion
						// Formerly assert 0x589
						const error = DataProcessingError.create(
```